meshtastic/mesh_interface.py

```
@@ -3093,243 +3540,306 @@ def _handle_queue_status_from_radio(
+            try:
+                node["position"] = self._fixup_position(node["position"])
+            except KeyError:
+                logger.debug("Node has no position key")
+
+            # no longer necessary since we're mutating directly in nodesByNum via _get_or_create_by_num
+            # self.nodesByNum[node["num"]] = node
+            if "user" in node:  # Some nodes might not have user/ids assigned yet
+                if "id" in node["user"]:
+                    # Keep nodes and nodesByNum mutation under the same lock
+                    # so readers never observe partially-updated node mappings.
+                    if self.nodes is not None:
```

Ruff (0.15.6)

[warning] Use a single `if` statement instead of nested `if` statements

(SIM102)